### PR TITLE
feat: add theme (fonts, colors, etc) to design kit

### PR DESCRIPTION
**Pull request by thiagodallacqua-hpe**
## Description

Ticket: WEB-817

The idea is to collect all information about the current theme structure into our design kit.

## Test Plan

go to the `/design` route, the theme info should be present.

Screen Shot 2023-05-18 at 15 44 48

Screen Shot 2023-05-23 at 11 36 51
Screen Shot 2023-05-23 at 11 36 46
Screen Shot 2023-05-23 at 11 36 37
Screen Shot 2023-05-23 at 11 36 29

## Commentary (optional)

With the latest implementation, the `Header` and the `Paragraph` components has new props:

```typescript
// at src/utils/fonts.ts
export const TypographySize = {
  default: 'default',
  L: 'L',
  S: 'S',
  XL: 'XL',
  XS: 'XS',
} as const;

// Header

interface Props {
  size?: TypographySize;
}

// Paragraph

interface Props {
  size?: TypographySize;
  type?: 'single line' | 'multi line';
  font?: 'code' | 'ui';
}
```

Each `size` has it's own `font-size` and `line-height` following design spec (check the screenshots above).

For the `Paragraph` there's a `type` prop, which differentiates the single line from the multi-line paragraphs, and the `font` prop which differentiates regular UI text to code text.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/pages/DesignKit.tsx [hunk 1]
```diff
@@ -56,6 +56,116 @@ import useConfirm, { voidPromiseFn } from '../components/kit/useConfirm';
 import css from './DesignKit.module.scss';
 import { CheckpointsDict } from './TrialDetails/F_TrialDetailsOverview';
 
+const themeStatus = [
```
Comment (ClaireNeveu) [inline webui/react/src/pages/DesignKit.tsx:59]: These shouldn't be defined in the demo page, they should be accessible to the rest of the design kit as enumerations.

File: webui/react/src/pages/DesignKit.tsx [hunk 2]
```diff
@@ -1734,6 +1845,111 @@ const TypographySection: React.FC = () => {
...
+      ))}
+    </AntDCard>
+  );
+  const iterateOverThemes = (themes: Array<string[]>, names: string[]) =>
+    themes.map((theme, idx) => renderColorComponent(theme, names[idx]));
+
+  return (
+    <ComponentSection id="Theme" title="Theme">
+      <AntDCard>
+        <p>Here are a quick demo of the current theme (colors, fonts, etc)</p>
```
Comment (ClaireNeveu) [inline webui/react/src/pages/DesignKit.tsx:1875]: "A variety of colors are available for use with the components in the UI Kit."

**Review by ClaireNeveu — CHANGES_REQUESTED** [review]

File: webui/react/src/utils/themes.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
+export enum Status {
```
Comment (ClaireNeveu) [inline webui/react/src/utils/themes.ts:1]: We've standardized on using "object as const" for enums, please bring this in-line.

File: webui/react/src/utils/themes.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  InteractiveOnActive = 'var(--theme-ix-on-active)',
+  InteractiveOnInactive = 'var(--theme-ix-on-inactive)',
+  InteractiveBorder = 'var(--theme-ix-border)',
+  InteractiveBorderStrong = 'var(--theme-ix-border-strong)',
+  InteractiveBorderWeak = 'var(--theme-ix-border-weak)',
+  InteractiveBorderActive = 'var(--theme-ix-border-active)',
+  InteractiveBorderInactive = 'var(--theme-ix-border-inactive)',
+}
+
+export enum Fonts {
```
Comment (ClaireNeveu) [inline webui/react/src/utils/themes.ts:119]: Enums should always be singular.

File: webui/react/src/utils/themes.ts [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  SuccessOnWeak = 'var(--theme-status-success-on-weak)',
+  Warning = 'var(--theme-status-warning)',
+  WarningStrong = 'var(--theme-status-warning-strong)',
+  WarningWeak = 'var(--theme-status-warning-weak)',
+  WarningOn = 'var(--theme-status-warning-on)',
+  WarningOnStrong = 'var(--theme-status-warning-on-strong)',
+  WarningOnWeak = 'var(--theme-status-warning-on-weak)',
+}
+
+export enum Backgrounds {
```
Comment (ClaireNeveu) [inline webui/react/src/utils/themes.ts:41]: Please put the colors in a `color` file and the fonts in a `font` file.

**Review by ClaireNeveu — CHANGES_REQUESTED** [review]
I don't see an enum for the font sizes?

File: webui/react/src/pages/DesignKit.tsx [hunk 6]
```diff
@@ -1734,6 +1746,120 @@ const TypographySection: React.FC = () => {
...
+            For general UI --theme-font-family
+          </p>
+          <p style={{ fontFamily: `${Fonts.Code}`, fontSize: '16px' }}>
+            For displaying code --theme-font-family-code
+          </p>
+        </AntDCard>
+        <AntDCard title="Font Sizing">
+          <div>
+            <div style={{ display: 'flex', flexDirection: 'column', marginBottom: '30px' }}>
+              <h4 style={{ marginBottom: '20px' }}>Header</h4>
```
Comment (ClaireNeveu) [inline webui/react/src/pages/DesignKit.tsx:1803]: We should be using the existing `Header` component.

File: webui/react/src/components/kit/Typography/Header.tsx [hunk 7]
```diff
@@ -2,8 +2,19 @@ import React from 'react';
 
 import css from './index.module.scss';
 
-const Header: React.FC<React.PropsWithChildren> = ({ children }) => {
-  return <h1 className={css.header}>{children}</h1>;
+interface Props {
+  className?: string;
```
Comment (ClaireNeveu) [inline webui/react/src/components/kit/Typography/Header.tsx:6]: UI Kit components should not take arbitrary classes or styles.

File: webui/react/src/components/kit/Typography/Paragraph.tsx [hunk 8]
```diff
@@ -1,7 +1,16 @@
 import React from 'react';
 
-const Paragraph: React.FC<React.PropsWithChildren> = ({ children }) => {
-  return <p>{children}</p>;
+interface Props {
```
Comment (ClaireNeveu) [inline webui/react/src/components/kit/Typography/Paragraph.tsx:3]: As above so here

File: webui/react/src/utils/fonts.ts [hunk 9]
```diff
@@ -0,0 +1,43 @@
+export const ThemeFont = {
+  Code: 'var(--theme-font-family-code)',
+  UI: 'var(--theme-font-family)',
+};
```
Comment (ClaireNeveu) [inline webui/react/src/utils/fonts.ts:4]: You need an `as const` on these so they infer correctly, you should also export a matching type. See other cases where we use enums as an example.

**Review by ClaireNeveu — CHANGES_REQUESTED** [review]

File: webui/react/src/utils/fonts.ts [hunk 10]
```diff
@@ -0,0 +1,50 @@
+import { ValueOf } from 'shared/types';
+
+export const ThemeFont = {
+  Code: 'var(--theme-font-family-code)',
+  UI: 'var(--theme-font-family)',
+} as const;
+
+export const Header = {
```
Comment (ClaireNeveu) [inline webui/react/src/utils/fonts.ts:8]: Are these actually used? It looks like you defined a separate enum in the components themselves.

Comment (ClaireNeveu) [reply webui/react/src/utils/fonts.ts:8]: Where would we use these other than the header? And if line height is something we want then why _doesn't_ header take them? (not clear to me that we do want that, was it a request from design?)

Comment (ClaireNeveu) [reply webui/react/src/utils/fonts.ts:8]: In that case what you have here doesn't accomplish what Alex asked for because it's not actually usable.

Comment (ClaireNeveu) [reply webui/react/src/utils/fonts.ts:8]: By CSS variables do you mean the `ParagraphSizes` you defined?

**Comment by hkang1** [comment]
**suggestion:**
- for the design kit layout of colors, can we put them in a `<Grid>` or something to make them more compact 🙏 ? They take up quite a bit of vertical space and there's a lot of space to the right on a typical browser size window.
- add a contrasting border to all the color boxes so even colors that are similar to the background it sits on will be visible

File: webui/react/src/utils/fonts.ts (hunk 10 above)
Comment (ClaireNeveu) [reply webui/react/src/utils/fonts.ts:8]: How do you use that with `Paragraph` or `Header`?

Comment (ClaireNeveu) [reply webui/react/src/utils/fonts.ts:8]: I can read the code. You just said that the enums in the ts code aren't the same as the css vars.

File: webui/react/src/utils/themes.ts [hunk 11]
```diff
@@ -0,0 +1,128 @@
+import { ValueOf } from 'shared/types';
```
Comment (ClaireNeveu) [inline webui/react/src/utils/themes.ts:1]: File should be color, not theme.

File: webui/react/src/utils/fonts.ts [hunk 12]
```diff
@@ -0,0 +1,16 @@
+import { ValueOf } from 'shared/types';
+
+export const ThemeFont = {
+  Code: 'var(--theme-font-family-code)',
+  UI: 'var(--theme-font-family)',
+} as const;
+
+export const TypographySizes = {
```
Comment (ClaireNeveu) [inline webui/react/src/utils/fonts.ts:8]: Should be singular.

File: webui/react/src/pages/DesignKit.tsx [hunk 13]
```diff
@@ -1821,6 +1834,122 @@ const TypographySection: React.FC = () => {
...
+      <AntDCard>
+        <Paragraph>
+          We have a variety of colors that are available for use with the components in the UI Kit.
+        </Paragraph>
+      </AntDCard>
+      {iterateOverThemes(
+        [themeStatus, backgrounds, stage, surface, float, overlay, brand, interactive],
+        ['Status', 'Background', 'Stage', 'Surface', 'Float', 'Overlay', 'Brand', 'Interactive'],
+      )}
+      <AntDCard title="Fonts">
```
Comment (ClaireNeveu) [inline webui/react/src/pages/DesignKit.tsx:1885]: These should be under `Typography`.

File: webui/react/src/pages/DesignKit.tsx [hunk 14]
```diff
@@ -1821,6 +1834,122 @@ const TypographySection: React.FC = () => {
...
+          </div>
+        ))}
+      </Grid>
+    </AntDCard>
+  );
+  const iterateOverThemes = (themes: Array<string[]>, names: string[]) =>
+    themes.map((theme, idx) => renderColorComponent(theme, names[idx]));
+
+  return (
+    <ComponentSection id="Theme" title="Theme">
```
Comment (ClaireNeveu) [inline webui/react/src/pages/DesignKit.tsx:1875]: Section should be `Color`.

**Review by ClaireNeveu — APPROVED** [review]
Check with @alexjdetermined That the actual colors/styles are correct. Looks good on my end.

**Review by hkang1 — APPROVED** [review]
Great work! Minor suggestions but nothing blocking.

**Comment by alexjdetermined** [comment]
1. --theme-status-inactive-weak/strong is flipped in order compared to the others.
2. Change border color around color blocks from "--theme-stage-on" to "--theme-surface-border"

**Review by alexjdetermined — APPROVED** [review]
LGTM!

**Pull request merged**